pkg/planner/core/rule_predicate_simplification.go

```diff
@@ -311,6 +347,83 @@ func pruneEmptyORBranches(sctx base.PlanContext, predicates []expression.Express
 	}
 }
 
+// shortCircuitANDORLogicalConstants simplifies logical expressions by performing short-circuit evaluation
+// based on the logical AND/OR nature of the predicate and constant truth/falsehood values.
+func shortCircuitANDORLogicalConstants(sctx base.PlanContext, predicate expression.Expression, orCase bool) (expression.Expression, bool) {
+	con, _ := predicate.(*expression.ScalarFunction)
+	args := con.GetArgs()
+	firstCondition, secondCondition := args[0], args[1]
+
+	// Recursively process first and second conditions
+	firstCondition, firstType := processCondition(sctx, firstCondition)
+	secondCondition, secondType := processCondition(sctx, secondCondition)
+
+	switch {
+	case firstType == truePredicate && orCase:
+		return firstCondition, true
+	case secondType == truePredicate && orCase:
+		return secondCondition, true
+	case firstType == falsePredicate && orCase:
+		return secondCondition, true
+	case secondType == falsePredicate && orCase:
+		return firstCondition, true
+	case firstType == truePredicate && !orCase:
+		return secondCondition, true
+	case secondType == truePredicate && !orCase:
+		return firstCondition, true
+	case firstType == falsePredicate && !orCase:
+		return firstCondition, true
+	case secondType == falsePredicate && !orCase:
+		return secondCondition, true
+	default:
+		if firstCondition != args[0] || secondCondition != args[1] {
+			finalResult := expression.NewFunctionInternal(sctx.GetExprCtx(), con.FuncName.L, con.GetStaticType(), firstCondition, secondCondition)
+			return finalResult, true
+		}
+		return predicate, false
+	}
+}
+
+// processCondition handles individual predicate evaluation for logical AND/OR cases
+// and returns the potentially simplified condition and its updated type.
+func processCondition(sctx base.PlanContext, condition expression.Expression) (expression.Expression, predicateType) {
+	applied := false
+	_, conditionType := FindPredicateType(sctx, condition)
+
+	if conditionType == orPredicate {
+		condition, applied = shortCircuitANDORLogicalConstants(sctx, condition, true)
+	} else if conditionType == andPredicate {
+		condition, applied = shortCircuitANDORLogicalConstants(sctx, condition, false)
+	}
+
+	if applied {
+		sctx.GetSessionVars().StmtCtx.SetSkipPlanCache("True/False predicate simplification is triggered")
```

Comment:
we can in the out most caller part set them only once, otherwise, like the embedded case will call it >1 times?